### ISSUE TITLE
last(::Tuple) should be type stable

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaLang/julia#19571]
This caught me by surprise:
```julia
julia> x=(1,2.0,"hi");

julia> @code_warntype first(x)
Variables:
  #self#::Base.#first
  t::Tuple{Int64,Float64,String}

Body:
  begin 
      return (Base.getfield)(t::Tuple{Int64,Float64,String},1)::Int64
  end::Int64

julia> @code_warntype(last(x))
Variables:
  #self#::Base.#last
  a::Tuple{Int64,Float64,String}

Body:
  begin 
      SSAValue(0) = (Base.nfields)(a::Tuple{Int64,Float64,String})::Int64
      return (Base.getfield)(a::Tuple{Int64,Float64,String},SSAValue(0))::Union{Float64,Int64,String}
  end::Union{Float64,Int64,String}
```

**Comment by Sacha0** [comment]
#19543

**Comment by JeffBezanson** [comment]
fixed by #19605.
